src/hoi4_world/states/hoi4_states_converter.cpp

```diff
@@ -364,6 +379,63 @@ std::tuple<int, int, int> ConvertIndustry(const float& total_factories,
 }
 
 
+std::map<int, int> CreateVictoryPoints(const std::set<int>& hoi4_provinces,
+    const mappers::Hoi4ToVic3ProvinceMapping& hoi4_to_vic3_province_mappings,
+    const std::map<std::string, std::string>& significant_provinces)
+{
+   std::map<int, int> victory_points;
+
+   std::set<std::string> applied_types;
+
+   for (const auto hoi4_province: hoi4_provinces)
+   {
+      const auto province_mapping = hoi4_to_vic3_province_mappings.find(hoi4_province);
+      if (province_mapping == hoi4_to_vic3_province_mappings.end())
+      {
+         // shouldn't be possible, since all hoi4 provinces come from the mappings, but just in case
+         continue;
+      }
+
+      for (const auto& vic3_province: province_mapping->second)
+      {
+         const auto& significant_province = significant_provinces.find(vic3_province);
+         if (significant_province == significant_provinces.end())
+         {
+            continue;
+         }
+
+         if (significant_province->second == "city" && !applied_types.contains("city"))
+         {
+            victory_points.emplace(hoi4_province, 5);
+            applied_types.insert("city");
+         }
+         else if (significant_province->second == "port" && !applied_types.contains("port"))
+         {
+            victory_points.emplace(hoi4_province, 4);
+            applied_types.insert("port");
+         }
+         else if (significant_province->second == "farm" && !applied_types.contains("farm"))
```

Comment:
Sanity/RP-wise, I'd shove farms to 1 point, and mines and woods (logging) leave at 2 points. Mines even maybe to 3.

Or maybe make it configurable in the future.

Reply:
This isn't intended to be balanced, but instead distinguish the different VPs within a state. This is generating what in the future will be debug VPs. #62 should make it an actual good feature.